### PR TITLE
add additional presubmit job for optional run (RHCS provider)

### PR DESCRIPTION
**Pull request by amalykhi**
Current presubmit job verifies only cluster creation and few fast checks after that. I would like to trigger full test run if I want to verify the changes in the tests (for example) or Dev wants to verify his changes more deeply. We have periodic job with suck checks but it can't be triggered from the PR IIUC

### CODE REVIEW

**Comment by robpblake** [comment]
@amalykhi Can you update the PR description to explain what this change is for? Do we have a JIRA issue for it? Just looking for some context.

cheers,
Rob

**Comment by amalykhi** [comment]
@robpblake  Current presubmit job verifies only cluster creation and few fast checks after that. I would like to trigger full test run if I want to verify the changes in the tests (for example) or Dev wants to verify his changes more deeply. We have periodic job with suck checks but it can't be triggered from the PR IIUC

**Comment by amalykhi** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-rhcs-rosa-sts-advanced-optional-critical-high-presubmit

**Comment by amalykhi** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-rhcs-rosa-sts-private-optional-critical-high-presubmit

**Comment by amalykhi** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-rhcs-rosa-sts-advanced-optional-critical-high-presubmit

**Comment by amalykhi** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-rhcs-rosa-sts-private-optional-critical-high-presubmit

File: ci-operator/jobs/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main-presubmits.yaml [hunk 1]
```diff
@@ -136,6 +136,86 @@ presubmits:
...
+    context: ci/prow/rosa-sts-advanced-optional-critical-high-presubmit
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+    labels:
+      ci-operator.openshift.io/cloud: aws
+      ci-operator.openshift.io/cloud-cluster-profile: aws-terraform-qe
+      ci.openshift.io/generator: prowgen
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-optional-critical-high-presubmit
```
Comment (nirarg) [inline ci-operator/jobs/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main-presubmits.yaml:154]: I'm afraid the name is too long :-( 
Is there a way to improve this?

File: ci-operator/jobs/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main-presubmits.yaml [hunk 2]
```diff
@@ -136,6 +136,86 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )rosa-sts-advanced-critical-high-presubmit,?($|\s.*)
+  - agent: kubernetes
```
Comment (nirarg) [inline ci-operator/jobs/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main-presubmits.yaml:139]: Can you please clarify what is the difference between this to the second job in this PR?

File: ci-operator/jobs/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main-presubmits.yaml (hunk 1 above)
Comment (nirarg) [reply ci-operator/jobs/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main-presubmits.yaml:154]: I know that I requested it to be more explicit
Just adding here another thought, need to find the middle :-(

File: ci-operator/jobs/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main-presubmits.yaml (hunk 2 above)
Comment (amalykhi) [reply ci-operator/jobs/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main-presubmits.yaml:139]: One for rosa-sts-pl (private) profile , another - for rosa-sts-ad (advanced)

File: ci-operator/jobs/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main-presubmits.yaml (hunk 1 above)
Comment (shellymiron) [reply ci-operator/jobs/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main-presubmits.yaml:154]: the thing is, we will need to change the other jobs to be aligned with this "imperative" term,
maybe not so convenient...

File: ci-operator/jobs/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main-presubmits.yaml (hunk 2 above)
Comment (nirarg) [reply ci-operator/jobs/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main-presubmits.yaml:139]: Ok, got it
You can resolve this

File: ci-operator/jobs/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main-presubmits.yaml (hunk 1 above)
Comment (nirarg) [reply ci-operator/jobs/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main-presubmits.yaml:154]: Ok, never mind, running the tests with shorter name

**Pull request merged**